code/tests.py

```diff
@@ -60,6 +60,11 @@ def test_nginx_generation(self):
         robots_nginx = json_to_nginx(self.robots_dict)
         self.assertEqualsFile("test_files/nginx-block-ai-bots.conf", robots_nginx)
 
+class TestRobotsNameCleaning(unittest.TestCase):
+    def test_clean_name(self):
+        from robots import clean_robot_name
+
+        self.assertEqual(clean_robot_name("Perplexity‑User"), "Perplexity-User")
```

Comment:
Another possibly simpler option would be to test this by adding the case to code/test-files/robots.json.

Reply:
I thought about that, and had a couple counterpoints:

1. any change to `code/test-files/robots.json` leads to a large diff as every test case needs its test-comparison-samples updated
2. `darkvisitors` no longer lists `Perplexity-Users` as a bot, so it wouldn’t be reliably tested by day-to-day execution. (And that was a complication as I was debugging the issue, haha, had to trace the issue back to a copy-paste from Open-AI’s docs page)
3. A natural user-maintenance-move would be to copy the “current” robots.json into there, and because of (1) and (2) it would be easy to lose track of my test case.

I figured a standalone test case would be clear, less likely to be accidentally omitted, and pretty easy to intentionally delete should it no longer be relevant.

Reply:
Yes, I buy that reasoning.